python/feldera/rest/client.py

```diff
@@ -0,0 +1,468 @@
+from typing import Optional
+import logging
+import time
+import json
+
+from feldera.rest.config import Config
+from feldera.rest.connector import Connector
+from feldera.rest.attached_connector import AttachedConnector
+from feldera.rest.program import Program
+from feldera.rest.pipeline import Pipeline
+from feldera.rest._httprequests import HttpRequests
+
+
+class Client:
+    """
+    A client for the Feldera HTTP API
+
+    A client instance is needed for every Feldera API method to know the location of
+    Feldera and its permissions.
+    """
+
+    def __init__(

```

Comment:
It may be a good idea to issue an HTTP request (e.g., list programs) to the manager, so that the user gets an error if the url or the api key are not valid early on.